mock-plugin/src/test/java/io/mosip/signup/plugin/mock/service/MockProfileRegistryPluginImplTest.java

```diff
@@ -0,0 +1,272 @@
+
+        Assert.assertEquals(status,ProfileCreateUpdateStatus.COMPLETED);
+    }
+
+    @Test
+    public void getProfile_withValidIndividualId_thenPass() throws ProfileException {
+        String individualId = "1234567890";
+        ReflectionTestUtils.setField(mockProfileRegistryPlugin, "getIdentityEndpoint","http://localhost:8080/");
+
+        Map<String, Object> verifiedData = new HashMap<>();
+        verifiedData.put("email","123@email.com");
+        verifiedData.put("password","123456");
+        verifiedData.put("UIN","1234567890");
+        verifiedData.put("individualId",individualId);
+
+        JsonNode mockIdentity = objectMapper.valueToTree(verifiedData);
+        ResponseWrapper<JsonNode> responseWrapper = new ResponseWrapper<>();
+        responseWrapper.setResponse(mockIdentity);
+        ResponseEntity<ResponseWrapper<JsonNode>> responseEntity=new ResponseEntity<>(responseWrapper, HttpStatus.OK);
+
+        Mockito.when(restTemplate.exchange(
+                "http://localhost:8080/"+individualId,
+                HttpMethod.GET,
+                null,
+                new ParameterizedTypeReference<ResponseWrapper<JsonNode>>() {
+                })).thenReturn(responseEntity);
+        ProfileDto profileDto= mockProfileRegistryPlugin.getProfile(individualId);
+        Assert.assertNotNull(profileDto);
+        Assert.assertEquals(profileDto.getIndividualId(),"1234567890");
+    }
+
+    @Test
+    public void getProfile_withErrorCodeAsIndividualIdFail_thenFail() throws ProfileException {
+        String individualId = "1234567890";
+        ReflectionTestUtils.setField(mockProfileRegistryPlugin, "getIdentityEndpoint","http://localhost:8080/");
+
+        Map<String, Object> verifiedData = new HashMap<>();
+        verifiedData.put("email","123@email.com");
+        verifiedData.put("password","123456");
+        verifiedData.put("UIN","1234567890");
+        verifiedData.put("individualId",individualId);
+
+        JsonNode mockIdentity = objectMapper.valueToTree(verifiedData);
+        ResponseWrapper<JsonNode> responseWrapper = new ResponseWrapper<>();
+        responseWrapper.setResponse(mockIdentity);
+        Error error = new Error();
+        error.setErrorCode("invalid_individual_id");
+        responseWrapper.setErrors(List.of(error));
+        ResponseEntity<ResponseWrapper<JsonNode>> responseEntity=new ResponseEntity<>(responseWrapper, HttpStatus.OK);
+
+        Mockito.when(restTemplate.exchange(
+                "http://localhost:8080/"+individualId,
+                HttpMethod.GET,
+                null,
+                new ParameterizedTypeReference<ResponseWrapper<JsonNode>>() {
+                })).thenReturn(responseEntity);
+
+        ProfileDto profileDto= mockProfileRegistryPlugin.getProfile(individualId);
+        Assert.assertNotNull(profileDto);
+        Assert.assertEquals(profileDto.getIndividualId(),individualId);
```

Comment:
Although it's a negative test case, assertion seems to be positive. Is this correct?

Reply:
Asserted the active status and hope it will look negative now